src/compiler/transformers/declarations/transpileDeclaration.ts

```diff
@@ -0,0 +1,118 @@
+    transformDeclarations,
+    TranspileDeclarationsOptions,
+    TranspileDeclarationsOutput,
+} from "../../_namespaces/ts";
+
+export function transpileDeclaration(sourceFile: SourceFile, transpileOptions: TranspileDeclarationsOptions): TranspileDeclarationsOutput {
+    const compilerOptions: CompilerOptions = {
+        ...transpileOptions.compilerOptions,
+        isolatedDeclarations: true,
+    };
+    const emitHost = {
+        getCurrentDirectory: () => transpileOptions.currentDirectory ?? ".",
+        getCanonicalFileName: createGetCanonicalFileName(!!compilerOptions.useCaseSensitiveFileNames),
+        useCaseSensitiveFileNames: () => !!compilerOptions.useCaseSensitiveFileNames,
+        getCompilerOptions: () => compilerOptions.compilerOptions,
+        getCommonSourceDirectory: () => ensureTrailingDirectorySeparator(transpileOptions.commonSourceDirectory ?? "."),
+    };
+    const emitResolver = createEmitDeclarationResolver(sourceFile);
+    const diagnostics: Diagnostic[] = [];
+    const transformationContext: IsolatedTransformationContext = {
+        ...nullTransformationContext,
+        kind: TransformationContextKind.IsolatedContext,
+        getCompilerOptions: () => compilerOptions,
+        addDiagnostic: diag => diagnostics.push(diag),
+        getEmitResolver: () => emitResolver,
+    };
+    const transformer = transformDeclarations(transformationContext);
+    const result = transformer(sourceFile);
+
+    const printer = createPrinter({
+        removeComments: compilerOptions.removeComments,
+        newLine: compilerOptions.newLine,
+        noEmitHelpers: true,
+        module: compilerOptions.module,
+        target: compilerOptions.target,
+        sourceMap: compilerOptions.declarationMap,
+        inlineSourceMap: compilerOptions.inlineSourceMap,
+        extendedDiagnostics: compilerOptions.extendedDiagnostics,
+        onlyPrintJsDocStyle: true,
+        omitBraceSourceMapPositions: true,
+    } as PrinterOptions);
+
+    const writer = createTextWriter(getNewLineCharacter(compilerOptions));
+    const declarationPath = getDeclarationEmitOutputFilePathWorker(
+        sourceFile.fileName,
+        compilerOptions,
+        emitHost.getCurrentDirectory(),
+        emitHost.getCommonSourceDirectory(),
+        emitHost.getCanonicalFileName,
+    );
+    const declarationMapPath = declarationPath + ".map";
+    const sourceMap = getSourceMapGenerator(declarationPath, declarationMapPath);
+    printer.writeFile(result, writer, sourceMap?.sourceMapGenerator);
+    if (sourceMap) {
+        if (!writer.isAtStartOfLine()) writer.writeLine();
+        writer.writeComment(sourceMap.sourceMappingURL);
+    }
+
+    return {
+        declaration: writer.getText(),
+        declarationPath,
+        declarationMap: sourceMap?.sourceMapGenerator.toString(),
+        declarationMapPath: sourceMap && declarationMapPath,
+        diagnostics,
+    };
+
+    // logic replicated from emitter.ts

```

Comment:
Is this something that can be easily pulled out?

Reply:
I feel it's not that easy because I only pulled out code in the emitter which is only relevant. 
I'll remove the confusing comment instead.